internal/metrics/metrics.go

```diff
@@ -127,3 +130,58 @@ func (s *ScheduleTimer) Mark(ctx context.Context, role, outcome string) {
 	metrics.FromContext(ctx).Histogram(
 		schedulingRetryTime, sparkRoleTag, outcomeTag, s.instanceGroupTag, s.retryTag).Update(now.Sub(s.lastSeenTime).Nanoseconds())
 }
+
+func ReportCrossZoneMetric(ctx context.Context, driverNodeName string, executorNodeNames []string, nodes []*v1.Node) {
+	executorNodesSet := make(map[string]interface{})
+	for _, n := range executorNodeNames {
+		executorNodesSet[n] = nil
+	}
+
+	zonesCounter := make(map[string]int64)
+	for _, n := range nodes {
+		if _, ok := executorNodesSet[n.Name]; ok {
+			executorZone, err := getNodeZone(n)
+			if err != nil {
+				return
+			}
+			increment(zonesCounter, executorZone)
```

Comment:
you can use `zonesCounter[executorZone]++` and remove the increment method

Reply:
ah nice, removed